word_search.py

```diff
@@ -0,0 +1,102 @@
+'''
+Created on 9 June 2017
+
+@author: Eva
+'''
+from queue import Queue
+
+class WordSearch:
+    
+    def __init__(self):
+        return
+        
+    def words_found_in_grid(self, rowNo, columnNo, grid, dictionary):
+        """Uses bfs to find valid words from a grid according to a given dictionary 
+        
+            Args:
+                rowNo: number of rows in the grid
+                columnNo: number of columns in the grid
+                grid: two-dimensional array of characters
+                dictionary: a dictionary object that holds the valid words and prefices
+            
+            Returns:
+                A list of words that were found on the grid
+        """
+        self.__pathsTaken = []
+        self.__foundWords = []
+        self.__words_to_search = Queue()
+        row = 0
+        column = 0
+        
+        self.__words_to_search.put([grid[column][row], column, row, []])
```

Comment:
This starting configuration seems to assume that:
* the word will start from the coordinates `(0, 0)`
* while you added the first character `grid[column][row]`, the `positions` array on the queue is empty, which means that you are not recording that you have already visited this cell.